PageViewController.swift

```diff
@@ -0,0 +1,211 @@
+        getStarted.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor, constant: -130).isActive = true
+        getStarted.isHidden = true
+        getStarted.layer.opacity = 0
+        
+    }
+    
+    @objc func close(_: UIButton) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+
+    @objc func skipClicked(_: UIButton) {
+        
+        for i in currentIndex...pages.count-1 {
+            setViewControllers([pages[i]], direction: .forward, animated: true, completion: nil)
+        }
+        pageControl.currentPage = pages.count - 1
+        
+    }
+    
+    @objc func nextClicked(_: UIButton){
+        print("Nexting now")
+        moveToNextPage()
+    }
+    
+    func moveToNextPage() {
+        
+        if currentIndex != pages.count - 1 {
+            setViewControllers([pages[currentIndex + 1]], direction: .forward, animated: true, completion: nil)
+        }
+        pageControl.currentPage += 1
+        
+    }
+    
+    func configurePageControl() {
+        let initialPage = 0
+        let page1 = StudyPactOnboardingViewController()
+        let page2 = StudyPactOnboarding2ViewController()
+        let page3 = StudyPactOnboarding3ViewController()
+        
+        // add the individual viewControllers to the pageViewController
+        self.pages.append(page1)
+        self.pages.append(page2)
+        self.pages.append(page3)
+        setViewControllers([pages[initialPage]], direction: .forward, animated: true, completion: nil)
+        
+        // pageControl
+        self.pageControl.frame = CGRect(x: 0, y: 0, width: 100, height: 100)
+        self.pageControl.currentPageIndicatorTintColor = UIColor.black
+        self.pageControl.pageIndicatorTintColor = UIColor.lightGray
+        self.pageControl.numberOfPages = self.pages.count
+        self.pageControl.currentPage = initialPage
+        self.pageControl.pageIndicatorTintColor = #colorLiteral(red: 0.337254902, green: 0.4352941176, blue: 0.7254901961, alpha: 1)
+        self.view.addSubview(self.pageControl)
+        
+        self.pageControl.translatesAutoresizingMaskIntoConstraints = false
+        self.pageControl.bottomAnchor.constraint(equalTo: self.view.bottomAnchor, constant: -65).isActive = true
+        self.pageControl.widthAnchor.constraint(equalToConstant: 140).isActive = true
+        self.pageControl.heightAnchor.constraint(equalToConstant: 20).isActive = true
+        self.pageControl.centerXAnchor.constraint(equalTo: self.view.centerXAnchor).isActive = true
+        self.pageControl.tintColor = UIColor(displayP3Red: 86/255, green: 111/255, blue: 185/255, alpha: 1)
+        self.pageControl.isUserInteractionEnabled = false
+    }
+    
+    func pageViewController(_ pageViewController: UIPageViewController, viewControllerBefore viewController: UIViewController) -> UIViewController? {
+        if let viewControllerIndex = self.pages.firstIndex(of: viewController) {
```

Comment:
`guard let viewControllerIndex = self.pages.firstIndex(of: viewController), viewControllerIndex != 0 else { return nil }` instead?